docs/using-cinder-csi-plugin.md

```diff
@@ -141,6 +141,41 @@ Currently, driver supports only one topology key: `topology.cinder.csi.openstack
 
 Note: `allowedTopologies` can be specified in storage class to restrict the topology of provisioned volumes to specific zones and should be used as replacement of `availability` parameter.
 
+### Example Snapshot Create and Restore
+
+Following prerequisite needed for volume snapshot feature to work.
+
+1. Enable `-feature-gates="VolumeSnapshotDataSource=true` in kube-apiserver
+2. Make Sure, your csi deployment contains external-snapshotter sidecar container, external-snapshotter sidecar container will create three crd's for snapshot management VolumeSnapshot,VolumeSnapshotContent, and VolumeSnapshotClass.       
+To deploy cinder csi driver and all required sidecars run
+```
+kubectl -f manifests/cinder-csi-plugin create
+```
+
+For Snapshot Creation and Volume Restore, please follow  below steps:
+
+* Create Storage Class, Snapshot Class and PVC    
+```
+$ kubectl -f examples/cinder-csi-plugin/example.yaml create
+```     
+* Verify that pvc is bounded
+``` 
+$ kubectl describe pvc
+```   
+* Create Snapshot of the PVC    
+```
+$ kubectl -f examples/cinder-csi-plugin/snapshotcreate.yaml create
+```       
+* Verify that snapshot is created    
+```
+$ kubectl get volumesnapshot 
+$ kubectl get volumesnapshotcontent
+```   
+* Restore volume from snapshot    
+```
+$ kubectl -f examples/cinder-csi-plugin/snapshotrestore.yaml create
+```
```

Comment:
could we add verify step here as well , the new volume is created with snapshot specified?

Reply:
done

Reply:
done